jenkins-dsl/cassandra_job_dsl_seed.groovy

```diff
@@ -370,72 +394,70 @@ cassandraBranches.each {
+                        retry(9, 5000)
                     }
-                    retry(9, 5000)
+                    failOnError(false)
                 }
-                failOnError(false)
-            }
-            matrixPostBuildScript {
-              buildSteps {
-                markBuildUnstable(false)
-                postBuildStep {
-                    executeOn('BOTH')
-                    stopOnFailure(false)
-                    results(['SUCCESS','UNSTABLE','FAILURE','NOT_BUILT','ABORTED'])
-                    buildSteps {
-                      shell {
-                        // docker needs to (soon or later) prune its volumes too, but that can only be done when the agent is idle
-                        // if the agent is busy, just prune everything that is older than maxJobHours
-                        command("""
-                            echo "Cleaning project…"; git clean -xdff ;
-                            echo "Cleaning processes…" ;
-                            if ! pgrep -af "cassandra-builds/build-scripts" ; then pkill -9 -f org.apache.cassandra. || echo "already clean" ; fi ;
-                            echo "Pruning docker…" ;
-                            if pgrep -af "cassandra-builds/build-scripts" ; then docker system prune --all --force --filter "until=${maxJobHours}h" || true ; else  docker system prune --all --force --volumes || true ;  fi;
-                            echo "Reporting disk usage…"; df -h ;
-                            echo "Cleaning tmp…";
-                            find . -type d -name tmp -delete 2>/dev/null ;
-                            find /tmp -type f -atime +2 -user jenkins -and -not -exec fuser -s {} ';' -and -delete 2>/dev/null || echo clean tmp failed ;
-                            echo "For test report and logs see https://nightlies.apache.org/cassandra/${branchName}/${jobNamePrefix}-artifacts/\${BUILD_NUMBER}/\${JOB_NAME}/"
-                        """)
-                      }
+                matrixPostBuildScript {
+                  buildSteps {
+                    markBuildUnstable(false)
+                    postBuildStep {
+                        executeOn('BOTH')
+                        stopOnFailure(false)
+                        results(['SUCCESS','UNSTABLE','FAILURE','NOT_BUILT','ABORTED'])
+                        buildSteps {
+                          shell {
+                            // docker needs to (soon or later) prune its volumes too, but that can only be done when the agent is idle
+                            // if the agent is busy, just prune everything that is older than maxJobHours
+                            command("""
+                                echo "Cleaning project…"; git clean -xdff ;
+                                echo "Cleaning processes…" ;
+                                if ! ( pgrep -xa docker || pgrep -af "cassandra-builds/build-scripts" ) ; then pkill -9 -f org.apache.cassandra. || echo "already clean" ; fi ;
```

Comment:
the `pgrep -xa docker ||` is needed before [CASSANDRA-18133](https://issues.apache.org/jira/browse/CASSANDRA-18133)